package/MDAnalysis/analysis/nucleicacids.py

```diff
@@ -183,50 +290,213 @@ class WatsonCrickDist(NucPairDist):
 
     Attributes
     ----------
-    times: numpy.ndarray
-        Simulation times for analysis.
     results.pair_distances: numpy.ndarray
         2D array of Watson-Crick basepair distances. First dimension is
         simulation time, second dimension contains the pair distances for
         each each entry pair in strand1 and strand2.
 
         .. versionadded:: 2.4.0
 
+    results.distances: numpy.ndarray
+        stored in a 2d numpy array with first index selecting the Residue pair, 
+        and the second index selecting the frame number
+    times: numpy.ndarray
+            Simulation times for analysis.
 
     Raises
     ------
-    ValueError
-        If the residues given are not amino acids
+    DeprecationWarning
+        If a lits of :class:`~MDAnalysis.core.groups.Residue` is given for
+        :attr:`strand1` of :attr:`strand2` instead of a
+        :class:`~MDAnalysis.core.groups.ResidueGroup`
     ValueError
         If the selections given are not the same length
-
+    ValueError:
+        An :class:`~MDAnalysis.core.groups.AtomGroup` in one of the strands not
+        a valid nucleic acid
+    ValueError
+        If a given residue pair from the provided strands returns an empty
+        :class:`~MDAnalysis.core.groups.AtomGroup` when selecting the atom
+        pairs used in the distance calculations
 
     .. versionchanged:: 2.5.0
        Accessing results by passing strand indices to :attr:`results` is
        was deprecated and is now removed as of MDAnalysis version 2.5.0. Please
        use :attr:`results.pair_distances` instead.
        The :attr:`results.times` was deprecated and is now removed as of
        MDAnalysis 2.5.0. Please use the class attribute :attr:`times` instead.
+
+    .. versionchanged:: 2.7.0
+        Deprecated the use of :attr:`List[Residue]` instead use
+        :class:`~MDAnalysis.core.groups.ResidueGroup`.
+
     """
 
-    def __init__(self, strand1: List[Residue], strand2: List[Residue],
+    def __init__(self, strand1: ResidueClass, strand2: ResidueClass,
                  n1_name: str = 'N1', n3_name: str = "N3",
                  g_name: str = 'G', a_name: str = 'A', u_name: str = 'U',
                  t_name: str = 'T', c_name: str = 'C',
                  **kwargs) -> None:
-        sel1: List[mda.AtomGroup] = []
-        sel2: List[mda.AtomGroup] = []
-        strand = zip(strand1, strand2)
+        
+        if isinstance(strand1, list) or isinstance(strand2, list):
+            raise DeprecationWarning("ResidueGroup should be used instead of giving a Residue list")
+            strand1: ResidueGroup = ResidueGroup(strand1)
+            strand2: ResidueGroup = ResidueGroup(strand2)
```

Comment:
This indentation does not look right. Please check.

Reply:
The indentation is fine, it appears wrong due to the `DeprecationWarning` going onto the next line. There is an error I overlooked thought, I used `raise` which caused a failure. I switched it to warn so that it just throws a warring then continues running the code. I'll also make sure the code continues working correctly after the warning to be safe.